sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -159,44 +159,142 @@ class AstBuilder extends DataTypeAstBuilder
     script
   }
 
-  private def visitCompoundBodyImpl(
-      ctx: CompoundBodyContext,
-      label: Option[String],
-      allowVarDeclare: Boolean,
-      labelCtx: SqlScriptingLabelContext,
-      isScope: Boolean): CompoundBody = {
-    val buff = ListBuffer[CompoundPlanStatement]()
-    ctx.compoundStatements.forEach(
-      compoundStatement => buff += visitCompoundStatementImpl(compoundStatement, labelCtx))
+  private def assertSqlState(sqlState: String): Unit = {
+    val sqlStateRegex = "^[A-Za-z0-9]{5}$".r
+    if (sqlStateRegex.findFirstIn(sqlState).isEmpty
+      || sqlState.startsWith("00")
+      || sqlState.startsWith("01")
+      || sqlState.startsWith("XX")) {
+      throw SqlScriptingErrors.invalidSqlStateValue(CurrentOrigin.get, sqlState)
+    }
+  }
+
+  private def visitConditionValueImpl(
+      ctx: ConditionValueContext,
+      handlerTriggers: HandlerTriggers): Unit = {
+    // Current element is SQLSTATE.
+    Option(ctx.sqlStateValue())
+      .foreach { sqlStateValueContext =>
+        val sqlState = sqlStateValueContext.getText.replace("'", "")
+        assertSqlState(sqlState)
+        handlerTriggers.addUniqueSqlState(sqlState)
+      }
 
-    val compoundStatements = buff.toList
+    // Current element is condition.
+    Option(ctx.multipartIdentifier())
+      .foreach { conditionContext =>
+        handlerTriggers.addUniqueCondition(conditionContext.getText)
```

Comment:
we need to distinguish qualified name `a.b` and name with special char `a.b`. The condition name should be `Seq[String]`.

Reply:
or we should add check here and fail if it's qualified.

Reply:
That's why `UnresolvedAttribute` uses `Seq[String]` to represent column names, as we can easily distinguish `Seq("a", "b")` and `Seq("a.b")`. The parser can recognize it by calling `visitMultipartIdentifier`.